### DISCUSSION TITLE
Full backups

### DISCUSSION THREAD

**Discussion by Sir-Alex-L** [mydumper/mydumper#1130] [Q&A]
Quick question, just considering this tool for full backups and restores to new server if needed. It's all in planning stages so I'm not actively testing at the moment but wanted to make sure my thoughts are correct.

Is this command enough to do full backups? And does it create separate files for each database? Or is it all one file?

The idea is that I won't even know what database names are going to be, but I'll have to be the one backing it all up. So I'm assuming regex of .* will backup everything. The idea of running this hourly with crontab. The database itself is 50GB.

```
today=$(date +"%Y-%m-%d")
mydumper --defaults-file /home/debian/mysql.mydumper.conf --regex '.*' --outputdir /var/MySQLbackup/${today} --build-empty-files --triggers --events  --routines --logfile /var/MySQLbackup/log --complete-insert --compress
```

**Comment by davidducos** [comment]
You should use:
```
--regex '^(?!(mysql\.))'
```
you don't need: `--build-empty-files`
Use -F and set to the size that you want, I think that 10 is a good starting point.
And -r 1000:10000:0
And if you are using Percona Server with GTID on, then you can use --no-locks
